task/task-pangolin-pangoro/src/service/fee.rs

```diff
@@ -0,0 +1,165 @@
+
+        let config_pangolin: ChainInfoConfig =
+            Config::restore_with_namespace(PangolinPangoroTask::NAME, "pangolin")?;
+        let config_pangoro: ChainInfoConfig =
+            Config::restore_with_namespace(PangolinPangoroTask::NAME, "pangoro")?;
+        let config_relay: RelayConfig = Config::restore(PangolinPangoroTask::NAME)?;
+
+        let (pangolin_chain, pangoro_chain) = (
+            config_pangolin
+                .to_chain_info_with_expect_signer(config_relay.signer_pangolin.clone())?,
+            config_pangoro.to_chain_info_with_expect_signer(config_relay.signer_pangoro.clone())?,
+        );
+
+        let _greet = Self::try_task(
+            &format!("{}-update-fee", PangolinPangoroTask::NAME),
+            async move {
+                let pangolin_client = pangolin_chain
+                    .to_substrate_relay_chain::<PangolinChain>()
+                    .await?;
+                let pangoro_client = pangoro_chain
+                    .to_substrate_relay_chain::<PangoroChain>()
+                    .await?;
+
+                let pangolin_signer = pangolin_chain.to_keypair::<PangolinChain>()?;
+                let pangoro_signer = pangoro_chain.to_keypair::<PangoroChain>()?;
+
+                let pangolin_api = PangolinApi::new(pangolin_client);
+                let pangoro_api = PangoroApi::new(pangoro_client);
+
+                while let Some(message) = rx.recv().await {
+                    let ret = match message {
+                        NewCostMessage::Pangolin(cost, decide) => {
+                            let fee = cost.saturating_mul(10);
+                            submit_pangolin_fee(&pangolin_api, pangolin_signer.clone(), fee, decide)
+                                .await
+                        }
+                        NewCostMessage::Pangoro(cost, decide) => {
+                            let fee = cost.saturating_mul(10);
+                            submit_pangoro_fee(&pangoro_api, pangoro_signer.clone(), fee, decide)
+                                .await
+                        }
+                    };
+                    if let Err(e) = ret {
+                        log::error!(
+                            target: PangolinPangoroTask::NAME,
+                            "Failed to update fee: {:?}",
+                            e
+                        );
+                    }
+                }
+                Ok(())
+            },
+        );
+        Ok(Self { _greet })
+    }
+}
+
+async fn submit_pangolin_fee(
+    api: &PangolinApi,
+    signer: <PangolinChain as TransactionSignScheme>::AccountKeyPair,
+    fee: common_primitives::Balance,
+    _decide: bool,
+) -> anyhow::Result<()> {
+    let signer_id: common_primitives::AccountId = (*signer.public().as_array_ref()).into();
+    let relayer = api.relayer(signer_id.clone()).await?;
+    if let Some(relayer) = relayer {
+        if fee <= relayer.fee {
+            return Ok(());
```

Comment:
11. If the current quotation is lower than the historical quotation, it will not be updated